tmt/cli.py

```diff
@@ -351,6 +351,21 @@ def main(
 @option(
     '--scratch', is_flag=True,
     help='Remove the run workdir before executing to start from scratch.')
+@option(
+    '--rerun', is_flag=True,
+    help='Rerun failed tests and update existing results.')
+@option(
+    '--force-rerun-test', multiple=True, metavar='[REGEXP|.]',
+    help="""
+         Regular expression to match test names that are supposed to be rerun
+         regardless of their previous result. Can be provided multiple times.
+         """)
+@option(
+    '--force-rerun-filter', 'force_rerun_filters', multiple=True, metavar='FILTER',
+    help="""
+         Apply advanced filter (see 'pydoc fmf.filter') to select tests that
+         are supposed to be rerun regardless of their previous result.
+         """)
```

Comment:
As for now options for choosing which tests should be executed are kept under the `tmt run test` subcommand.

    tmt run test --name NAME
    tmt run test --filter NAME
    tmt run test --link RELATION:TARGET

Would it make sense to put the new options for selecting failed-only tests there as well? It could possibly play nicely with the recently added `--again` option as well:

    tmt run
    tmt run --last --again
    tmt run --last execute --again
    tmt run --last --again test --failed-only
    tmt run --last --again test --name 'setup|specific-test'
    tmt run --last --again test --filter tag:setup --name --failed-only

In this way the test-selection options would be at one location.

Reply:
@KwisatzHaderach ^

Reply:
the problem with --again is that it keeps the workdir as is, ending with doubled results, I'm not sure if the expectation for it is to double the results (run has 10 tests, you run it with again and now you have 20 results) or if that is a bug due to no testing done on the feature

to move filtering (filter, name, failed-only) under test makes sense, I just have to find a way how to access it 😄

Reply:
I believe, the problem with duplicated results has been already [resolved](https://github.com/teemtee/tmt/pull/2528/files#diff-458194650925d6c3ea27ee8a599db67a78a23dbb0b353cddc9e2b8387b54d993R719) in the original pull request:

```
> cd tests/execute/basic/data/

> tmt run -av -u execute plan -n bea
/var/tmp/tmt/run-031
Found 1 plan.
...
    execute
        how: tmt
            00:00:00 fail /test/beakerlib/bad (on default-0) [1/4]
            00:00:00 pass /test/beakerlib/good (on default-0) [2/4]
            00:00:00 warn /test/beakerlib/need (on default-0) [3/4]
            00:00:00 errr /test/beakerlib/weird (on default-0) (beakerlib: TestResults FileError) [4/4]
        summary: 4 tests executed

total: 1 test passed, 1 test failed, 1 warn and 1 error

> tmt run -l execute -v --again
/var/tmp/tmt/run-031

/plan/beakerlib
    execute
        how: tmt
            00:00:00 fail /test/beakerlib/bad (on default-0) [1/4]
            00:00:00 pass /test/beakerlib/good (on default-0) [2/4]
            00:00:00 warn /test/beakerlib/need (on default-0) [3/4]
            00:00:00 errr /test/beakerlib/weird (on default-0) (beakerlib: TestResults FileError) [4/4]
        summary: 4 tests executed

total: 1 test passed, 1 test failed, 1 warn and 1 error
```